tree/ntuple/v7/src/RField.cxx

```diff
@@ -1649,6 +1652,131 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
 
 //------------------------------------------------------------------------------
 
+TClass *ROOT::Experimental::RField<TObject>::GetClass()
+{
+   return TClass::GetClass("TObject");
+}
+
+std::size_t ROOT::Experimental::RField<TObject>::GetOffsetOfMember(const char *name)
+{
+   for (auto dataMember : ROOT::Detail::TRangeStaticCast<TDataMember>(*GetClass()->GetListOfDataMembers())) {
+      if (strcmp(name, dataMember->GetName()) == 0)
+         return dataMember->GetOffset();
+   }
+   throw RException(R__FAIL("invalid data member"));
+}
+
+ROOT::Experimental::RField<TObject>::RField(std::string_view fieldName)
+   : ROOT::Experimental::RFieldBase(fieldName, "TObject", ENTupleStructure::kRecord, false /* isSimple */)
+{
+   R__ASSERT(GetClass()->GetClassVersion() == 1);
```

Comment:
Why is this needed?

Reply:
That is, are we expecting TObject to ever change class version?

Reply:
We should not expect `TObject` to ever change and thus we should cost a run-time check for it.  (A regular `assert` costing only in debug build would be fine).